### ISSUE TITLE
Some thoughts about use existing code from other projects

### ISSUE THREAD

**Issue by rogerc66** [ZhengPeiRu21/mod-playerbots#147]
I tried Npcbot and I think  playerbot still the best idea of populate server with live bot and the ultimate fun of the way to go, However, the npc fighting and manage scripts are really well implemented, Is it possible to migrate that part of code to playerbot so that it could save so much time for development?

**Comment by heyitsbench** [comment]
Probably not, given one uses creatures and the other uses characters as the vessel of bots.

**Comment by htc16** [comment]
I don't think so either. And I also used npc bot, but I think I prefer playerbot.

**Comment by htc16** [comment]
Guys, I've been testing some things and I even informed Zhang about it. I noticed that in the playerbot source there are many `.csv` extensions that are not being created automatically.

I even verified that there is already a download for some types of `.csv`. Here: https://github.com/celguar/mangosbot-bots/blob/878e02b2b612de8cf032fb69d433bbe2cff6b833/playerbot/aiplayerbot.conf.dist.in.wotlk#L520

`
AiPlayerbot.AllowedLogFiles = "bot_events.csv,bot_movement.csv,bot_pathfinding.csv,creatures.csv,crop.csv,Deaths.csv,gos.csv,load_map_grid.csv,pathfind_attempt.csv,pathfind_attempt_point.csv,pathfind_result.csv,player_location.csv,scan.csv,strategy.csv,travel_map.csv,travelNodes.csv,travelPaths.csv,unload_grid.csv,unload_obj.csv,vmangoslines.csv,zones.csv"
`

Another point that I noticed, especially the `player_location.csv` file, it made the bot much more responsive, to the point that I did quests in a party with them.

![image](https://user-images.githubusercontent.com/58153179/210081175-a0549a36-9ce3-4ea3-a872-905d3a6e0870.png)

If you can test that too, that would be interesting.

**Comment by rogerc66** [comment]
@htc16 very interesting finding, I will try.

**Comment by htc16** [comment]
I already have the answer for these .csv files, I'll share it with you!

**Comment by htc16** [comment]
I just got the answers from the creator of these files I downloaded.

**Comment by htc16** [comment]
These `.csv` files just provide information about what is happening with the bots, how many levels the bots are gaining, how hard the server is working, how many bots are online, etc.

**Comment by htc16** [comment]
![image](https://user-images.githubusercontent.com/58153179/210084612-3e335e28-e58d-414b-bdd3-81c23770019a.png)
![image (1)](https://user-images.githubusercontent.com/58153179/210084397-64dad231-0ac0-4b29-a1ea-c1ea2a87688a.png)
![image
![image (3)](https://user-images.githubusercontent.com/58153179/210084408-f9cfc55a-448f-402b-9ed0-b6b158497ae0.png)
 (2)](https://user-images.githubusercontent.com/58153179/210084401-2c01e161-9d30-46fd-bcfa-5136ef184e50.png)
![image (4)](https://user-images.githubusercontent.com/58153179/210084414-594b44ce-b0f6-4480-bfac-cd7838e08000.png)
![image (5)](https://user-images.githubusercontent.com/58153179/210084421-06fa9e7f-2f86-4a71-8777-38cef5cde949.png)

These .csv files just provide information about what is happening with the bots, how many levels the bots are gaining, how hard the server is working, how many bots are online, etc.

they work like metrics!
@ZhengPeiRu21

**Comment by rogerc66** [comment]
looks really cool

**Comment by Dreathean** [comment]
@rogerc66 please close this issue so we can clean up the high amount of old issues clogging this, I think Bench's answer still stands that they can't copy over the creature combat scripts for the playerbots. If something new comes up we can open up a new issue but for now this discussion is old and done.
